src/averageAges.js

```diff
@@ -55,7 +86,18 @@ function calculateWomenAverageAge(people, withChildren) {
  * @return {number}
  */
 function calculateAverageAgeDiff(people, onlyWithSon) {
-  // write code here
+  const filteredTab = (onlyWithSon)
+    ? people.filter(isMan).filter(isChild)
+    : people.filter(isChild);
+
+  function isChild(person) {
```

Comment:
The `isChild` function should be moved outside of `calculateAverageAgeDiff` to keep the main function cleaner, and to avoid redefining it on each function call. 😇